### PR TITLE
Countries table header

### PR DESCRIPTION
**Pull request by Globalika**
added header to countries table
added header image
Screenshot 2021-12-03 at 15 34 12
Screenshot 2021-12-03 at 15 35 12

### CODE REVIEW

**Review by khrystyna-sb — COMMENTED** [review]
whether the title should be transparent during scrolling?

**Comment by Globalika** [comment]
yep my mistake. wrong background color. will change it

**Review by khrystyna-sb — APPROVED** [review]
cool

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        label.font = .systemFont(ofSize: Constants.headerLabelFont)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        return label
+    }()
+
+    override func layoutSubviews() {
+        backgroundView = UIView(frame: self.bounds)
+        backgroundView?.backgroundColor = Constants.headerBackgroundColor
+    }
```
Comment (apolwork) [inline CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:32]: - Where is `super` call?
- You will override background view each time when `layoutSubviews` called

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        let label = UILabel()
+        label.font = .systemFont(ofSize: Constants.headerLabelFont)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        return label
+    }()
+
+    override func layoutSubviews() {
+        backgroundView = UIView(frame: self.bounds)
+        backgroundView?.backgroundColor = Constants.headerBackgroundColor
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:31]: Why `contentView` can not handle `backgroundColor`?

File: CountriesOfTheWorld/CountriesOfTheWorld/ViewControllers/CountriesController.swift [hunk 3]
```diff
@@ -49,8 +52,17 @@ extension CountriesController {
...
     }
 
-    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
-        return "Countries"
+    override func tableView(_ tableView: UITableView,
+                            viewForHeaderInSection section: Int) -> UIView? {
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: CountriesHeaderView.identifier)
+                as? CountriesHeaderView else { return UITableViewHeaderFooterView() }
+        header.imageView.image = UIImage(named: Constants.headerImageName)
+        header.headerLabel.text = Constants.headerLabelText
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/ViewControllers/CountriesController.swift:60]: can this be setup in `CountriesHeaderView`?

Comment (apolwork) [inline CountriesOfTheWorld/CountriesOfTheWorld/ViewControllers/CountriesController.swift:60]: If text and image constant, you could set in CountriesHeaderView

**Review by apolwork — CHANGES_REQUESTED** [review]

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift (hunk 2 above)
Comment (Globalika) [reply CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:31]: was not thinking about it. yes looks better

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift (hunk 1 above)
Comment (Globalika) [reply CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:32]: forgot about super. will add it

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        label.font = .systemFont(ofSize: Constants.headerLabelFont)
+        label.text = Constants.headerLabelText
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        return label
+    }()
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        contentView.backgroundColor = Constants.headerBackgroundColor
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:33]: line #33 can be moved to `init()` or `configureTableHeader()`

File: CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        label.font = .systemFont(ofSize: Constants.headerLabelFont)
+        label.text = Constants.headerLabelText
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        return label
+    }()
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+    }
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/Headers/CountriesHeader.swift:33]: what for you need this func? it is empty...

**Review by ambrusha — APPROVED** [review]

**Pull request merged**